packages/driver/src/dom/elements/find.ts

```diff
@@ -224,55 +223,91 @@ export const getElements = ($el) => {
   return els
 }
 
-export const getContainsSelector = (text, filter = '', options: {
-  matchCase?: boolean
-} = {}) => {
-  const $expr = $.expr[':']
+/*
```

Comment:
I hadn't intended to do anything here in this PR / initiative, but I kept running into bugs where the old logic would break with multiple .contains() statements + aliases in the same test. This was happening because I was only calling `getContainsSelector` once, rather than each time a retry occurred, and not cleaning it up properly; the method I thought was just returning a string was also modifying the global jquery Object!

The old `getContainsSelector` had side effects, making it hard to use safely. The new one is purely functional - it takes arguments and returns a string. It also performs better, since we register these three handlers once, and allow jquery to compile / cache them, rather than adding the `:cy-contains()` expression with internal state on each retry loop.